src/cosalib/ostree.py

```diff
@@ -0,0 +1,216 @@
+# Copyright 2018-2020 Red Hat, Inc
+# Licensed under the new-BSD license (http://www.opensource.org/licenses/bsd-license.php)
+
+import json
+import logging as log
+import os
+import subprocess
+import tarfile
+import tempfile
+import json
+import gi
+import sys
+
+gi.require_version('OSTree', '1.0')
+from gi.repository import (
+    Gio,
+    OSTree
+)
+
+
+from cosalib.cmdlib import (
+    get_basearch,
+    load_json,
+    sha256sum_file,
+    run_verbose,
+    write_json)
+from cosalib.build import _Build
+
+DEFAULT_OSTREE_META_SIZE_PERCENT = 5
+DEFAULT_OSTREE_PADDING_PERCENT = 35
+DEFAULT_OSTREE_INODE_SIZE = 512
+DEFAULT_OSTREE_BLOCK_SIZE = 4069
```

Comment:
I see...I've removed `estimate_ostree_commit_size` from the class, and then changed `OSTreeBuild` to `BuildOSTree` to remove confusion. I need to fix this method up before merging.

Reply:
Now though it looks odd because basically *everything* else you've changed is part of an object...so I guess if you want to keep going that way we should move this into `_BuildOSTree` too?